network/src/network.rs

```diff
@@ -738,65 +738,83 @@ impl<T: ExitHandler> NetworkService<T> {
         exit_handler: T,
     ) -> Self {
         let config = &network_state.config;
-        // == Build special protocols
-
-        // TODO: how to deny banned node to open those protocols?
-        // Ping protocol
-        let ping_interval = Duration::from_secs(config.ping_interval_secs);
-        let ping_timeout = Duration::from_secs(config.ping_timeout_secs);
-
-        let ping_network_state = Arc::clone(&network_state);
-        let (ping_handler, ping_controller) =
-            PingHandler::new(ping_interval, ping_timeout, ping_network_state);
-        let ping_meta = SupportProtocols::Ping.build_meta_with_service_handle(move || {
-            ProtocolHandle::Callback(Box::new(ping_handler))
-        });
+        // == Build p2p service struct
+        let mut protocol_metas = protocols
+            .into_iter()
+            .map(CKBProtocol::build)
+            .collect::<Vec<_>>();
 
-        // Discovery protocol
-        let addr_mgr = DiscoveryAddressManager {
-            network_state: Arc::clone(&network_state),
-            discovery_local_address: config.discovery_local_address,
-        };
-        let disc_meta = SupportProtocols::Discovery.build_meta_with_service_handle(move || {
-            ProtocolHandle::Callback(Box::new(DiscoveryProtocol::new(
-                addr_mgr,
-                config
-                    .discovery_announce_check_interval_secs
-                    .map(Duration::from_secs),
-            )))
-        });
+        // == Build special protocols
 
-        // Identify protocol
+        // Identify is a core protocol, user cannot disable it via config
         let identify_callback =
             IdentifyCallback::new(Arc::clone(&network_state), name, version.clone());
         let identify_meta = SupportProtocols::Identify.build_meta_with_service_handle(move || {
             ProtocolHandle::Callback(Box::new(IdentifyProtocol::new(identify_callback)))
         });
+        protocol_metas.push(identify_meta);
 
-        // Feeler protocol
-        let feeler_meta = SupportProtocols::Feeler.build_meta_with_service_handle({
-            let network_state = Arc::clone(&network_state);
-            move || ProtocolHandle::Both(Box::new(Feeler::new(Arc::clone(&network_state))))
-        });
+        // TODO: how to deny banned node to open those protocols?
```

Comment:
The premise of opening any protocol is to establish a connection. This step will check the ban list, if i remember correctly

Reply:
removed misleading comment line, please review again, thanks.